### PR TITLE
Add FT support for sending and receiving tokens

### PR DESCRIPTION
**Pull request by judith-Near**
Screen Shot 2022-06-09 at 11 59 02 AM

1) displays fungible tokens that were sent to the multisafe (either by another multisafe, or by the wallet account itself)
2) ability to send fungible tokens that you own to other multisafes or wallet accounts

### CODE REVIEW

**Review by marcinbodnar — COMMENTED** [review]
Looks good. Few issues with implementation below. 

Also, I encounter an issue while transferring FT.
When I'm sending wNEAR to the account that doesn't have a wNEAR yet, then wired things start to happen:
 - first, the tx seems to be successful, but when I enter Dashboard in multisafe the transaction is not confirmed - the approve checkbox was checked.
  - when I try it a second time, it shows that the transaction gas was exceeded (in Wallet)
  - after the third time it's successful.

File: src/store/multisafe/thunks/onTransferTokens.js [hunk 1]
```diff
@@ -38,19 +46,43 @@ const signTxByLedger = async (
...
+      // to do:
+      // : await signTxByLedger(fungibleTokensService, contract, withApprove, recipientId, amount, multisafeId, state, actions, contractName);
+  }
 
-  isNearWallet
-    ? addTransferRequest(contract, withApprove, recipientId, amount)
-    : await signTxByLedger(contract, withApprove, recipientId, amount, multisafeId, state, actions);
+  const fungibleTokens = state.multisafe.general.fungibleTokens;
+
+  const updatedTokens = await Promise.all(await fungibleTokens.map(async ({contractName}) => {
```
[the remark this answers is not in the document]
Comment (judith-Near) [reply src/store/multisafe/thunks/onTransferTokens.js:77]: implemented the refresh balance button and removed this. good catch about how the flow transfer is only finished under certain circumstances

File: src/store/multisafe/thunks/onTransferTokens.js [hunk 2]
```diff
@@ -38,19 +46,43 @@ const signTxByLedger = async (
...
+  // token is assumed to be 'near' if alternate token is not given
+  if (isNearTransaction) {
+    isNearWallet
+      ? addTransferNearRequest(contract, withApprove, recipientId, amount)
+      : await signTxByLedger(contract, withApprove, recipientId, amount, multisafeId, state, actions);
+  }
+  else {
+    isNearWallet
+      && await fungibleTokensService.addTransferRequest({multisafeContract: contract, withApprove, recipientId, amount: parseOtherAmount(token, amount), contractName: token.contractName})
+      // to do:
```
[the remark this answers is not in the document]
Comment (judith-Near) [reply src/store/multisafe/thunks/onTransferTokens.js:71]: separately

**Comment by judith-Near** [comment]
@agileurbanite feedback has been addressed, marcin is going to review and then should be good to merge. this PR allows FT transfer using the near wallet, but not with ledger yet -- using it with ledger will be updated in an additional PR.

**Comment by agileurbanite** [comment]
@judith-Near probably needs a rebase

File: src/store/multisafe/thunks/onTransferTokens.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (judith-Near) [reply src/store/multisafe/thunks/onTransferTokens.js]: kidding, has been addressed in this PR :)

**Review by marcinbodnar — APPROVED** [review]
LGTM

**Pull request merged**